gcloud-java-dns/src/main/java/com/google/gcloud/dns/testing/OptionParsers.java

```diff
@@ -0,0 +1,258 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.dns.testing;
+
+import com.google.api.services.dns.model.Change;
+import com.google.api.services.dns.model.ManagedZone;
+import com.google.api.services.dns.model.Project;
+import com.google.api.services.dns.model.ResourceRecordSet;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Utility helpers for LocalDnsHelper.
+ */
+class OptionParsers {
+
+  static Map<String, Object> parseListZonesOptions(String query) {
+    Map<String, Object> options = new HashMap<>();
+    if (query != null) {
+      String[] args = query.split("&");
+      for (String arg : args) {
+        String[] argEntry = arg.split("=");
+        switch (argEntry[0]) {
+          case "fields":
+            // List fields are in the form "managedZones(field1, field2, ...),nextPageToken"
+            String replaced = argEntry[1].replace("managedZones(", ",");
+            replaced = replaced.replace(")", ",");
+            // we will get empty strings, but it does not matter, they will be ignored
+            options.put("fields", replaced.split(","));
+            break;
+          case "dnsName":
```

Comment:
Fine as is, but just wanted to mention an alternative:
```java
case "dnsName":
case "pageToken":
case "maxResults":
  options.put(argEntry[0], argEntrty[1]);
  break;
```

Also, because it doesn't seem that you care if query has an unrecognized option (is that intentional?) you could have done this way:
```java
for (String arg : args) {
  String[] argEntry = arg.split("=");
  if ("fields".equals(argEntity[0])) {
    ...
  } else if (argsEntry.length > 1) {
    options.put(argsEntry[0], argsEntry[1]);
  }
}     
```